modules/gdscript/gdscript_parser.cpp

```diff
@@ -4129,7 +4129,7 @@ String GDScriptParser::DataType::to_string() const {
 			return class_type->fqcn;
 		case SCRIPT: {
 			if (is_meta_type) {
-				return script_type->get_class_name().operator String();
+				return script_type != nullptr ? script_type->get_class_name().operator String() : "";
```

Comment:
I'm not sure how much we need metatypes in error messages, but there is a difference. You can see it here:

```gdscript
var value1: Node
var type1 := Node

var value2: MyClass
var type2 := MyClass

var test: int

test = value1 # Value of type "Node" cannot be assigned to a variable of type "int".
test = type1 # Value of type "GDScriptNativeClass" cannot be assigned to a variable of type "int".

# 4.0.2:
test = value2 # Value of type "MyClass" cannot be assigned to a variable of type "int".
test = type2 # Value of type "GDScript" cannot be assigned to a variable of type "int".
# master after #75605:
test = value2 # Value of type "MyClass" cannot be assigned to a variable of type "int".
test = type2 # Value of type "MyClass" cannot be assigned to a variable of type "int".
```

For `kind == SCRIPT` I couldn't find an example, it looks like `preload()` also returns `kind == CLASS`. Maybe we should remove `is_meta_type` check in other cases of this method, or maybe #75605 was a mistake.

Reply:
`kind == SCRIPT` will only happen for different scripting languages. Maybe we can test this with C# (we probably should, as the `SCRIPT` kind was never really tested).

Reply:
So how do we process to get this merged / the crash scenario fixed?

Reply:
@vnen Bump!

Reply:
In my opinion, the check will not hurt. Not sure if this fixes the crash in the Language Server. Also perhaps we should also revert #75605 and fix the bug described there in a different way.

Reply:
Well this would definitely fix a potential nullptr dereference crash at least.

I'll merge for now, and leave it to you folks to open further issues/PRs if this needs further work.